### PR TITLE
Fixing #211

### PR DESCRIPTION
**Pull request by dublx**
Fixing https://github.com/mzabriskie/axios/issues/211

### CODE REVIEW

File: lib/adapters/http.js [hunk 1]
```diff
@@ -3,16 +3,23 @@
...
 var zlib = require('zlib');
 var pkg = require('./../../package.json');
 var Buffer = require('buffer').Buffer;
 
 // Resolve or reject the Promise based on the status
 function settle(resolve, reject, response) {
-  (response.status >= 200 && response.status < 300 ?
+  var minCode = 200;
+  var maxCode = 300;
+  if (response.config.hasOwnProperty('maxRedirects')) {
```
Comment (mzabriskie) [inline lib/adapters/http.js]: Why change the `maxCode`? Anything in the 300 range should be followed, unless `maxRedirects` occurs, in which case I assume that would be considered an error. Thoughts?

Comment (nickuraltsev) [inline lib/adapters/http.js]: I agree with @mzabriskie . On a side note, I'm going to submit a PR that adds support for a config option that allows specifying status codes for which an error should be thrown.

**Comment by nickuraltsev** [comment]
@dublx Could you please update README.md?

File: lib/adapters/http.js (hunk 1 above)
Comment (nickuraltsev) [inline lib/adapters/http.js]: See #308

**Comment by dublx** [comment]
Hi, I'm not too experienced with PR's.. do I update README and then submit new one?
Anyway, reason I change max code was because then the promise would be rejected.
I will change it then:  check if `maxRedirects === 0`, the idea being that there should be no redirects, but also no promise rejection. Agree?

**Comment by nickuraltsev** [comment]
You can just push a new commit to the same branch. It will be automatically added to your PR. 

As for max code, my suggestion is to not change the status code error range at all. If you don't want axios to reject the promise for 3XX responses, you will be able to do that via `validateStatus` config option when #308 is merged.

**Comment by nickuraltsev** [comment]
@dublx Please let me know if you need any help

**Comment by cspray** [comment]
@nickuraltsev Hi, I've really enjoyed using axios in the chatbot I'm creating for my company. I need this PR to support some needed functionality and currently plan to fork my own copy until this is available. If it is ok with @dublx I would be comfortable with cleaning this branch up to not change the status code error range, update the README and fix any conflicts. Alternatively I could simply create a new PR.

In short, I'd like to see the work I'm about to do be put back into axios to share with others and so I don't have to maintain my own fork. Please let me know if I can be assistance.

**Comment by nickuraltsev** [comment]
@cspray Yes, I think the right way to deal with this is to merge @dublx commits into your own branch, make necessary changes, and submit a new PR. If you could do that, it would be great. Thank you!

**Comment by cspray** [comment]
@nickuraltsev  Will do!

**Comment by barbraz** [comment]
Hi guys @dublx @nickuraltsev @cspray @mzabriskie, is there a chance that this PR would be merged soon? I'm really looking forward to it :) Can I help with it somehow?
Cheers :)

**Comment by nickuraltsev** [comment]
Merged
